### DISCUSSION TITLE
Add ICMP scan to Prefixes (like php-ipam)

### DISCUSSION THREAD

**Discussion by CheapVibe** [netbox-community/netbox#16671] [Ideas]
- Add Scan/Discover button for each prefix
- If button checked - background script will run once a day (or any desired interval from plugin page) and will ICMP scan the specific subnet (nmap if you will)
- New IP Addresses discovered will be created as active status
- Resolve new IP Address discovered PTR and store it in 'DNS name' field
- Existing IP Addresses that will not reply to ICMP - their status will change to dead/deprecated/inactive
- New IP addresses Description field will be 'Automatically Discovered <date>'
- Both new and inactive hosts addresses 'Last updated' field will updated
- Inactive hosts will be completely removed from NetBox after 30 days or any desired period from plugin page
- For each subnet add 'Scan Now' button. popup will display the new IP Addresses (with PTR resolve as hostname) and approve to add each one of the IP addresses

php-ipam discover for specific subnet:
![image](https://github.com/netbox-community/netbox/assets/121057304/b243c4db-390b-4498-82e5-450f21cef7d0)

php-ipam inactive hosts widget:
![image](https://github.com/netbox-community/netbox/assets/121057304/ffc06dd0-c85b-4f26-a0be-e04a167ce2e7)

php-ipam scan now subnet popup:
![image](https://github.com/netbox-community/netbox/assets/121057304/7086cf62-8975-4082-b40d-c802a4410290)

This will be a significant added value to Netbox!

Thank you 👍

**Comment by jeremystretch** [comment]
Please see our FAQ: [Why Doesn't NetBox Scan for IPs?](https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#why-doesnt-netbox-scan-for-ips)

**Reply by CheapVibe** [reply]
the simple answer for this is to create additional nodes on the remote sites to scan the "local" site (same as in php-ipam). the agent will push the data by ansible to the root node API.
can request this as a feature?
